File: Backend/Pro/3rd place Yurii Holenko/app/models/game.rb
class Game < ApplicationRecord
  belongs_to :round
  belongs_to :home_team, class_name: 'Team'
  belongs_to :guest_team, class_name: 'Team'
  has_one :game_result

  scope :finished, -> { where(finished: true) }
  scope :by_tournament, lambda { |tournament_id|
    joins(:round).where(rounds: { tournament_id: tournament_id })
  }

  scope :by_team, lambda { |team_id|
    where('guest_team_id = ? OR home_team_id = ?', team_id, team_id)
  }

  def score
    game_result ? "#{game_result.home_team_goals}-#{game_result.guest_team_goals}" : "-"
  end
end

File: Backend/Pro/1st place Dima Misik/SportChallenge.Core/Models/Game.cs
namespace SportChallenge.Core.Models
{
    public class Game : IIDentifiable
    {
        public int Id { get; set; }
        public Team HomeTeam { get; set; }
        public Team GuestTeam { get; set; }
        public Round Round { get; set; }
        public int HomeTeamId { get; set; }
        public int GuestTeamId { get; set; }
        public GameResult Result { get; set; }
    }
}
